fabric-renderer-registries-v1/src/main/java/net/fabricmc/fabric/api/client/rendereregistry/v1/RegisterFeatureRendererCallback.java

```diff
@@ -0,0 +1,81 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.rendereregistry.v1;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+import net.minecraft.client.render.entity.LivingEntityRenderer;
+import net.minecraft.client.render.entity.feature.Deadmau5FeatureRenderer;
+import net.minecraft.client.render.entity.feature.FeatureRenderer;
+import net.minecraft.client.render.entity.model.EntityModel;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+/**
+ * Called when {@link FeatureRenderer feature renderers} for a {@link LivingEntityRenderer living entity renderer} are registered.
+ *
+ * <p>Feature renderers are typically used for rendering additional objects on an entity, such as armor, an elytra or {@link Deadmau5FeatureRenderer Deadmau5's ears}.
+ * This callback lets developers add additional feature renderers for use in entity rendering.
+ * Listeners should filter out the specific entity renderer they want to hook into, usually through {@code instanceof} checks.
+ * Once listeners find a suitable entity renderer, they should register their feature renderer via the acceptor.
+ *
+ * <p>For example, to register a feature renderer for a player model, the example below may used:
+ * <blockquote><pre>
+ * RegisterFeatureRendererCallback.EVENT.register((entityType, entityRenderer, acceptor) -> {
+ * 	if (entityRenderer instanceof PlayerEntityModel) {
+ * 		acceptor.accept(new MyFeatureRenderer((PlayerEntityModel) entityRenderer));
+ *    }
+ * });
+ * </pre></blockquote>
+ */
+@FunctionalInterface
+@Environment(EnvType.CLIENT)
+public interface RegisterFeatureRendererCallback {
+	Event<RegisterFeatureRendererCallback> EVENT = EventFactory.createArrayBacked(RegisterFeatureRendererCallback.class, callbacks -> (entityType, entityRenderer, acceptor) -> {
+		for (RegisterFeatureRendererCallback callback : callbacks) {
+			callback.registerFeatureRenderers(entityType, entityRenderer, acceptor);
+		}
+	});
+
+	/**
+	 * Called when feature renderers may be registered.
+	 *
+	 * @param entityType     the entity type of the renderer
+	 * @param entityRenderer the entity renderer
+	 * @param acceptor       the accepted used to register feature renderers
+	 */
+	void registerFeatureRenderers(EntityType<? extends LivingEntity> entityType, LivingEntityRenderer<?, ?> entityRenderer, FeatureAcceptor acceptor);
+
+	final class FeatureAcceptor {
```

Comment:
imo you should just make this an interface and the impl creates an instance of `FeatureAcceptor` when it calls the event instead.

Reply:
I'll consider it. Could even implement it onto living entity renderer itself if I went for an interface (but that would not be part of docs and an impl detail

Reply:
Exposing this in the api defeats the point of abstraction; users don't need to know how this interface is implemented.